### PR TITLE
Updates ion-rust to rc10

### PR DESCRIPTION
**Pull request by zslayton**
### Description of changes:

----
_**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**_

### CODE REVIEW

File: src/bin/ion/commands/inspect.rs [hunk 1]
```diff
@@ -456,12 +455,25 @@ impl<'a, 'b> IonInspector<'a, 'b> {
...
+                    raw_eexp.opcode_and_address_span().bytes(),
+                ),
+                IonBytes::new(
+                    BytesKind::TrailingLength,
+                    raw_eexp.length_prefix_span().bytes(),
+                ),
+                IonBytes::new(
+                    BytesKind::ArgumentEncodingBitmap,
+                    raw_eexp.bitmap_span().bytes(),
+                ),
```
Comment (zslayton) [inline src/bin/ion/commands/inspect.rs:476]: 🪧 More sophisticated support for e-expressions--the address, length, and bitmap now have their own colors.

File: src/bin/ion/commands/inspect.rs [hunk 2]
```diff
@@ -494,9 +511,66 @@ impl<'a, 'b> IonInspector<'a, 'b> {
                     use MacroExprKind::*;
                     match invocation.kind() {
                         EExp(eexp_arg) => self.inspect_eexp(depth + 1, eexp_arg)?,
-                        EExpArgGroup(_) => todo!("e-exp arg groups"),
+                        EExpArgGroup(arg_group) => {
+                            self.inspect_eexp_arg_group(depth + 1, arg_group)?
```
Comment (zslayton) [inline src/bin/ion/commands/inspect.rs:515]: 🪧 There's now basic support for arg groups. E-expressions in the arg group do not yet expand.

File: src/bin/ion/hex_reader.rs [hunk 3]
```diff
@@ -120,54 +119,60 @@ impl<R: Read> Read for HexReader<R> {
     }
 }
 
-#[test]
```
Comment (zslayton) [inline src/bin/ion/hex_reader.rs:123]: 🪧 No logic changes here, I just moved the tests into a `#[cfg(test)]` module so they could share conditionally included imports.

**Review by zslayton — COMMENTED** [review]
🗺️ PR tour 🧭

**Review by jobarr-amzn — APPROVED** [review]
Thanks!

**Review by tgregg — APPROVED** [review]

**Pull request merged**